state/execution_test.go

```diff
@@ -0,0 +1,329 @@
+package state
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	abci "github.com/tendermint/abci/types"
+	"github.com/tendermint/basecoin/types"
+	"github.com/tendermint/go-crypto"
+)
+
+//States and Stores for tests
+var (
+	store                                 types.KVStore
+	state                                 *State
+	accsFoo, accsBar, accsFooBar, accsDup []types.PrivAccount
+	chainID                               string = "test_chain_id"
+)
+
+func makeAccs(secrets []string) (accs []types.PrivAccount) {
+
+	for _, secret := range secrets {
+		privAcc := types.PrivAccountFromSecret(secret)
+		privAcc.Account.Balance = types.Coins{{"mycoin", 1000}}
+		accs = append(accs, privAcc)
+	}
+	return accs
+}
+
+func acc2State(accs []types.PrivAccount) {
+	for _, acc := range accs {
+		state.SetAccount(acc.Account.PubKey.Address(), &acc.Account)
+	}
+}
+
+//each tx input signs the tx bytes
+func signSend(tx *types.SendTx, accs []types.PrivAccount) {
+	signBytes := tx.SignBytes(chainID)
+	for i, _ := range tx.Inputs {
+		tx.Inputs[i].Signature = crypto.SignatureS{accs[i].Sign(signBytes)}
+	}
+}
+
+//turn a list of accounts into basic list of transaction inputs
+func accs2TxInputs(accs []types.PrivAccount) []types.TxInput {
+	var txs []types.TxInput
+	for _, acc := range accs {
+		tx := types.NewTxInput(
+			acc.Account.PubKey,
+			types.Coins{{"mycoin", 5}},
+			1)
+		txs = append(txs, tx)
+	}
+	return txs
+}
+
+//turn a list of accounts into basic list of transaction outputs
+func accs2TxOutputs(accs []types.PrivAccount) []types.TxOutput {
+	var txs []types.TxOutput
+	for _, acc := range accs {
+		tx := types.TxOutput{
+			acc.Account.PubKey.Address(),
+			types.Coins{{"mycoin", 4}}}
+		txs = append(txs, tx)
+	}
+	return txs
+}
+
+//reset the store/state/Inputs
+func reset() {
+	accsFoo = makeAccs([]string{"foo"})
+	accsBar = makeAccs([]string{"bar"})
+	accsFooBar = makeAccs([]string{"foo", "bar"})
+	accsDup = makeAccs([]string{"foo", "foo", "foo"})
+
+	store = types.NewMemKVStore()
+	state = NewState(store)
+	state.SetChainID(chainID)
+}
+
+func TestGetInputs(t *testing.T) {
+	assert := assert.New(t)
+
+	//nil submissions
+	reset()
+	acc, res := getInputs(nil, nil)
+	assert.False(res.IsErr(), "getInputs: error on nil submission")
+	assert.Zero(len(acc), "getInputs: accounts returned on nil submission")
+
+	//test getInputs for registered, non-registered account
+	reset()
+	txs := accs2TxInputs(accsFoo)
+	_, res = getInputs(state, txs)
+	assert.True(res.IsErr(), "getInputs: expected to getInput from registered Input")
```

Comment:
I think the message needs to change (its the same as the next one, but with different outcomes)